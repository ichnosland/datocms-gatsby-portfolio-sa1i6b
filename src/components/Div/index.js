/**
*
* Div
*
*/

import styled, { css } from 'styled-components';
import media from 'style/mediainjector';

const Div = styled.div`
  display: ${(props) => props.display ? props.display : 'block'};
  position: ${(props) => props.position ? props.position : 'relative'};
  width: ${(props) => props.width};
  max-width: ${(props) => props.maxWidth ? props.maxWidth : '100%'};
  height: ${(props) => props.height};
  color: ${(props) => props.color ? props.color : 'inherit'};
  text-align: ${(props) => props.align};
  float: ${(props) => props.float ? props.float : 'none'};
  margin: ${(props) => props.margin ? props.margin : '0'};
  padding: ${(props) => props.padding ? props.padding : '0'};
  border: ${(props) => props.border ? props.border : 'none'};
  border-width: ${(props) => props.borderwidth};
  border-style: ${(props) => props.borderstyle};
  border-color: ${(props) => props.bordercolor};
  background-color: ${(props) => props.bgColor ? props.bgColor : 'transparent'};
  border-radius: ${(props) => props.radius};
  overflow: ${(props) => props.overflow ? props.overflow : 'visible'};
  user-select: ${(props) => props.userSelect};
  pointer-events: ${(props) => props.pointerEvents};
  z-index: ${(props) => props.zindex};
  ${(props) => props.nobreak && css`
    page-break-inside: avoid;
  `}
  ${media.print`
    ${({ noPrint }) => noPrint && css`
        display:none;
    `}
  `}
`;

export default Div;
