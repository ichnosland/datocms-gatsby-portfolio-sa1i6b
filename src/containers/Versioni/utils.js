import {
  createStepFromEsercizi,
  calcolaSoluzioni,
  cleanTesto,
} from 'common/esercizi';

/**
 * Dato in input l'array con i periodi della versione, restituisce
 * un oggetto contenente gli ID dei singoli periodi e un oggetto in cui,
 * per ogni id di ogni periodo, l'array con gli step associati
 *
 * @param {array} periodi della versione
 * @returns {object} oggetto con chiave periodiIds e stepPerPeriodi
 * dove periodiIds è un array contenente le chiavi dei periodi incompleti
 * e stepPerPeriodi è un oggetto con chiave la pk del periodo e come
 * valore l'array delle pk degli step ancora da eseguire
 * {
 *  periodiIds: [pk1, pk2, pk3]
 *  stepPerPeriodi: {
 *    [pkPeriodo1]: [pkStep1, pkStep2, pkStep3]
 *    [pkPeriodo2]: [pkStep1, pkStep2, pkStep3]
 *  }
 * }
 */
export const calcolaStrutturaStep = (periodi) => (
  periodi.reduce((acc, singoloPeriodo) => {
    acc.periodiIds.push(singoloPeriodo.periodo_id);
    acc.stepPerPeriodi[singoloPeriodo.periodo_id] = singoloPeriodo.periodo.map(
      (step) => (step.id)
    );
    return acc;
  }, { periodiIds: [], stepPerPeriodi: {} })
);

/**
 * Dati in input l'array dei periodi e la matrice di esecuzione
 * con le risposte fornite dall'utente, calcola quali sono i
 * periodi e gli step ancora da risolvere.
 *
 * @param {array} periodi della versione
 * @param {object} matriceEsecuzione risposte fornite dall'utente
 * @returns {object} oggetto con chiave periodiIncompletiPks e stepPerPeriodi
 * dove periodiIncompletiPks è un array contenente le chiavi dei periodi incompleti
 * e stepPerPeriodi è un oggetto con chiave la pk del periodo e come
 * valore l'array delle pk degli step ancora da eseguire
 * {
 *  periodiIncompletiPks: [pk1, pk2, pk3]
 *  stepDaEseguire: {
 *    [pkPeriodo1]: [pkStep1, pkStep2, pkStep3]
 *    [pkPeriodo2]: [pkStep1, pkStep2, pkStep3]
 *  }
 * }
 */
export const calcolaPeriodiDaEseguire = (periodi = [], risposteFornite = {}) => {
  const strutturaStep = calcolaStrutturaStep(periodi);
  return strutturaStep.periodiIds.reduce((acc, periodoId) => {
    strutturaStep.stepPerPeriodi[periodoId].forEach((stepId) => {
      if (!risposteFornite[periodoId] || !risposteFornite[periodoId][stepId]) {
        acc.periodiIncompletiPks = [...new Set([...acc.periodiIncompletiPks, periodoId])];
        if (!acc.stepDaEseguire[periodoId]) {
          acc.stepDaEseguire[periodoId] = [];
        }
        acc.stepDaEseguire[periodoId] = [...new Set([...acc.stepDaEseguire[periodoId], stepId])];
      }
    });
    return acc;
  }, { periodiIncompletiPks: [], stepDaEseguire: {} });
};

/**
 * Dati in input l'elenco delle pk degli step da tenere e l'elenco degli step
 * che compongono un periodo, restituisce l'array con gli step da eseguire
 *
 * @param {array} pksStepDaEseguire elenco delle pk degli step da eseguire
 * @param {array} stepsPeriodo elenco degli step presenti nel periodo
 */
export const calcolaContenutoPeriodoDaEseguire = (pksStepDaEseguire, stepsPeriodo) => (
  stepsPeriodo.filter((step) => (pksStepDaEseguire.indexOf(step.id) > -1))
);

/**
 * Dato in input un array con i periodi, ricava il testo
 * da tradurre della versione usando come testo buono da mostrare
 * per ogni periodo, il campo 'testo_principale' del primo elemento
 * del primo step. Questo per richiesta della redazione.
 *
 * @param {array} periodi periodi della versione
 */
export const calcolaTestoIntroduttivoPeriodi = (periodi) => (
  periodi.map((periodo, key) => ({
    testoDaTradurre: cleanTesto(
      periodo.periodo[0].elementi[0].testo_principale,
      'leavepunct'
    ).replace(/_/g, ' '),
    idPeriodo: periodo.periodo_id,
    keyPeriodo: key,
  }))
);

/**
 * Prende in input l'array degli step di un singolo periodo della
 * versione e lo trasforma in un array di periodi in cui gli esercizi
 * vengono esaminati e suddivisi in elementi di testo / esercizi
 * Questo perché in un esercizio posso avere, almeno in via teorica
 * più elementi di testo, e se presumo che la tipologia sia portata
 * sempre dall'elemento con chiave [1], potrei avere errori nell'analisi
 * dei dati
 * Questo mi serve anche per unificare la struttura dei singoli
 * esercizi da dare in pasto all'esecuzione
 *
 * @param {array} arrayStepDaConvertire elenco degli step da convertire
 */
export const versionePeriodoToSteps = (arrayStepDaConvertire) => (
  arrayStepDaConvertire.periodo.map((singoloPeriodo) => ({
    periodoId: arrayStepDaConvertire.periodo_id,
    id: singoloPeriodo.id,
    ...createStepFromEsercizi(singoloPeriodo.elementi)[0],
  }))
);

/**
 * Calcola il testo della traduzione fornita dall'utente. Per convenzione
 * assemblo la traduzione fornite alle risposte di tipo O e di tipo U.
 * Per accordo con la redazione uso solo il tipo O e il tipo U.
 * Se l'utente ha fornito a quella domanda una risposta sbagliata, utilizzo
 * il testo di default
 *
 * Per convenzione decisa con la redazione uso le domande di tipo U o O per
 * riassemblare il testo tradotto dall'utente
 *
 * @param {object} contenutoPeriodo contiene l'oggetto con i dati del periodo da analizzare
 * @param {object} contenutoPeriodo.periodo contiene l'array dei singoli step del periodo
 * @param {object} rispostaUtente oggetto contenente le risposte dell'utente
 * @returns {string} HTML con traduzione fornita dallo studente
 */
export const calcolaTraduzioneUtentePeriodo = (contenutoPeriodo, rispostaUtente = {}) => (
  contenutoPeriodo.periodo.filter((singoloStep) => (
    ['U', 'O'].indexOf(singoloStep.esercizi[0].tipo) > -1
  )).reduce((acc, singoloStep) => {
    const soluzioneStep = calcolaSoluzioni(singoloStep.testi, singoloStep.esercizi, 1)[0];

    switch (singoloStep.esercizi[0].tipo) {
      case 'U':
        singoloStep.esercizi.forEach((esercizio, key) => {
          if (esercizio.testo_principale_pre) {
            acc.push(esercizio.testo_principale_pre);
          }

          if ((rispostaUtente[singoloStep.id] || {}).stato === 'C') {
            acc.push(`<span class="corrected">${rispostaUtente[singoloStep.id].answer[key]}</span>`);
          } else {
            acc.push(`<span class="suggested">${soluzioneStep[key][0]}</span>`);
          }

          if (esercizio.testo_principale_post) {
            acc.push(esercizio.testo_principale_post);
          }
        });
        break;
      case 'O':
        if ((rispostaUtente[singoloStep.id] || {}).stato === 'C') {
          acc.push(`<span class="corrected">${rispostaUtente[singoloStep.id].readable}</span>`);
        } else {
          acc.push(`<span class="suggested">${soluzioneStep}</span>`);
        }
        break;
      /* istanbul ignore next */
      default: break;
    }
    return acc;
  }, []).join(' '));
