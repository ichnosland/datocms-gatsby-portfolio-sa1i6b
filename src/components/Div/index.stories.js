import React from 'react';
import { storiesOf } from '@storybook/react';
import { withNotes } from '@storybook/addon-notes';
import { withKnobs, text, boolean, color } from '@storybook/addon-knobs';
import Div from './index';


const nota = 'La props nobreak assegna \'page-break-inside: avoid;\' \n quella noPrint \'display:none;\' in stampa';

storiesOf('Div', module)
  .addDecorator(withKnobs)
  .addWithJSX('Div', withNotes(nota)(() => (
    <Div
      display={text('display', 'block')}
      position={text('position', 'relative')}
      width={text('width', 'auto')}
      maxWidth={text('maxWidth', '100%')}
      height={text('height', '')}
      color={color('color', 'inherit')}
      align={text('align (tex-align)', '')}
      float={text('float', 'none')}
      margin={text('margin', '0')}
      padding={text('padding', '0')}
      radius={text('radius', '')}
      bgColor={color('bgColor', 'transparent')}
      border={text('border', '')}
      borderwidth={text('borderwidth', '')}
      borderstyle={text('borderstyle', '')}
      bordercolor={color('bordercolor', '')}
      overflow={text('overflow', 'visible')}
      userSelect={text('userSelect', 'auto')}
      pointerEvents={text('pointerEvents', 'auto')}
      zindex={text('zindex', 'auto')}
      nobreak={boolean('nobreak', false)}
      noPrint={boolean('noPrint', true)}
    >
      <p>Ciao, sono all&apos;interno di un Div. Usa le props per personalizzarlo :)</p>
    </Div>
  )));
