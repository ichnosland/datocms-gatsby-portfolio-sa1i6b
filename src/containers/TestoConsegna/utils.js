/**
*
* ElementiTesto
*
*/

import React from 'react';
import 'katex/dist/katex.min.css';
import TeX from '@matejmazur/react-katex';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import { Lessico, Contesto } from 'components/Text';
import { HtmlStyles } from 'style/HtmlStylesVerifiche';

export const HtmlWrap = styled(Div)`
  width: 100%;
  ${HtmlStyles}
  ${({ inline }) => inline && css`
    display: inline;
  `}
`;

export const generaElementiTesto = (props) => {
  const {
    elementiTesto = [],
    hasLessico = false,
    lessicoRefs,
    toggleAperti,
  } = props;

  return elementiTesto.map((t) => {
    switch (t.type) {
      default:
        return t.content.replace(/_/g, ' ');
      case 'context':
        return (
          <Contesto key={`contesto_${t.id}`}>
            {generaElementiTesto({
              elementiTesto: t.children,
              hasLessico,
              lessicoRefs,
              toggleAperti,
            })}
          </Contesto>
        );
      case 'latex':
        return (<TeX key={`${t.type}_${t.id}`} math={t.content} />);
      case 'image':
        return (
          <HtmlWrap key={`${t.type}_${t.id}`} dangerouslySetInnerHTML={{ __html: t.content }} /> // eslint-disable-line
        );

      case 'html':
        return (
          <HtmlWrap key={`${t.type}_${t.id}`} dangerouslySetInnerHTML={{ __html: t.content }} inline /> // eslint-disable-line
        );

      case 'lessico':
        if (!hasLessico || !lessicoRefs || !toggleAperti) {
          return t.parola.replace(/_/g, ' ');
        }

        return (
          <Lessico
            key={`lessico_block_${t.id}`}
            ref={(input) => { lessicoRefs[`lessico_block_${t.id}`] = input; }}
            onClick={() => toggleAperti(t)}
          >
            {t.parola.replace(/_/g, ' ')}
          </Lessico>
        );
    }
  });
};
