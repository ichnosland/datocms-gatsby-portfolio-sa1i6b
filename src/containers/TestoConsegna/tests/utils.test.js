import React from 'react';
import { shallow, mount } from 'enzyme';

import Div from 'components/Div';

import { generaElementiTesto, HtmlWrap } from '../utils';

const mockTesti = [
  {
    type: 'lessico',
    parola: 'Hi_nc',
    inputLessicale: {
      traduzione: 'da qui',
      focus: false,
      forma: 'hinc',
      lemma: 'hinc',
      tipo: null,
      tabelle: null,
    },
    id: 'lessico_0_Hinc',
  }, {
    type: 'text',
    content: ' te_xt ',
    id: 'text_1',
  }, {
    type: 'lessico',
    parola: 'ille',
    inputLessicale: {
      traduzione: 'quello',
      focus: false,
      forma: 'ille',
      lemma: 'ille',
      tipo: 'p',
      tabelle: null,
    },
    id: 'lessico_2_ille',
  }, {
    type: 'context',
    id: 'context_34',
    children: [{
      type: 'lessico',
      parola: 'aeneumque',
      inputLessicale: {
        traduzione: 'di bronzo',
        focus: false,
        forma: 'aeneumque',
        lemma: 'aeneus',
        tipo: 'a',
        tabelle: null,
      },
      id: 'lessico_0_aeneumque',
    }, {
      type: 'text',
      content: 'testo',
      id: 'text_1',
    }],
  }, {
    type: 'text',
    content: '.',
    id: 'text_35',
  }, {
    type: 'html',
    content: '<strong>blocco html</strong>',
    id: 'text_36',
  }, {
    type: 'image',
    content: '<img src="immagine.png" />',
    id: 'image_36',
  }, {
    type: 'latex',
    content: '4+6',
    id: 'latex_45',
  }];



describe('generaElementiTesto', () => {
  it('contenuti misti senza lessico', () => {
    const elementiGenerati = generaElementiTesto({
      elementiTesto: mockTesti,
    });

    expect(elementiGenerati).toEqual([
      'Hi nc',
      ' te xt ',
      'ille',
      expect.any(Object),
      '.',
      expect.any(Object),
      expect.any(Object),
      expect.any(Object),
    ]);

    // contesto
    expect(elementiGenerati[3].props).toEqual({
      children: ['aeneumque', 'testo'],
    });
    expect(elementiGenerati[3].key).toBe('contesto_context_34');
    expect(elementiGenerati[3].type.displayName).toBe('Text__Contesto');

    // HtmlWrap html
    expect(elementiGenerati[5].props).toEqual({
      dangerouslySetInnerHTML: {
        __html: '<strong>blocco html</strong>',
      },
      inline: true,
    });
    expect(elementiGenerati[5].key).toBe('html_text_36');
    expect(elementiGenerati[5].type.displayName).toBe('utils__HtmlWrap');

    // HtmlWrap image
    expect(elementiGenerati[6].props).toEqual({
      dangerouslySetInnerHTML: {
        __html: '<img src="immagine.png" />',
      },
    });
    expect(elementiGenerati[6].key).toBe('image_image_36');
    expect(elementiGenerati[6].type.displayName).toBe('utils__HtmlWrap');

    // TeX
    expect(elementiGenerati[7].props).toEqual({
      math: '4+6',
    });
    expect(elementiGenerati[7].key).toBe('latex_latex_45');
  });

  it('elementiTesto undefined', () => {
    const elementiGenerati = generaElementiTesto({
      elementiTesto: undefined,
    });

    expect(elementiGenerati).toEqual([]);
  });

  it('contenuti misti con lessico', () => {
    const elementiGenerati = generaElementiTesto({
      elementiTesto: mockTesti,
      lessicoRefs: {},
      hasLessico: true,
      toggleAperti: () => { },
    });

    expect(elementiGenerati).toEqual([
      expect.any(Object),
      ' te xt ',
      expect.any(Object),
      expect.any(Object),
      '.',
      expect.any(Object),
      expect.any(Object),
      expect.any(Object),
    ]);

    // lessico 1
    expect(elementiGenerati[0].props).toEqual({
      children: 'Hi nc',
      onClick: expect.any(Function),
    });
    expect(elementiGenerati[0].key).toBe('lessico_block_lessico_0_Hinc');
    expect(elementiGenerati[0].type.displayName).toBe('Text__Lessico');

    // lessico 2
    expect(elementiGenerati[2].props).toEqual({
      children: 'ille',
      onClick: expect.any(Function),
    });
    expect(elementiGenerati[2].key).toBe('lessico_block_lessico_2_ille');
    expect(elementiGenerati[2].type.displayName).toBe('Text__Lessico');

    // contesto
    expect(elementiGenerati[3].props).toEqual({
      children: [expect.any(Object), 'testo'],
    });
    expect(elementiGenerati[3].key).toBe('contesto_context_34');
    expect(elementiGenerati[3].type.displayName).toBe('Text__Contesto');

    // lessico nel contesto
    const lessico = elementiGenerati[3].props.children[0];
    expect(lessico.props).toEqual({
      children: 'aeneumque',
      onClick: expect.any(Function),
    });
    expect(lessico.key).toBe('lessico_block_lessico_0_aeneumque');
    expect(lessico.type.displayName).toBe('Text__Lessico');

    // HtmlWrap html
    expect(elementiGenerati[5].props).toEqual({
      dangerouslySetInnerHTML: {
        __html: '<strong>blocco html</strong>',
      },
      inline: true,
    });
    expect(elementiGenerati[5].key).toBe('html_text_36');
    expect(elementiGenerati[5].type.displayName).toBe('utils__HtmlWrap');

    // HtmlWrap image
    expect(elementiGenerati[6].props).toEqual({
      dangerouslySetInnerHTML: {
        __html: '<img src="immagine.png" />',
      },
    });
    expect(elementiGenerati[6].key).toBe('image_image_36');
    expect(elementiGenerati[6].type.displayName).toBe('utils__HtmlWrap');

    // TeX
    expect(elementiGenerati[7].props).toEqual({
      math: '4+6',
    });
    expect(elementiGenerati[7].key).toBe('latex_latex_45');
  });

  it('testo ref Lessico', () => {
    const toggleAperti = jest.fn();
    const lessicoRefs = {};
    const elementiGenerati = generaElementiTesto({
      elementiTesto: [{
        type: 'lessico',
        parola: 'Hi_nc',
        inputLessicale: {
          traduzione: 'da qui',
          focus: false,
          forma: 'hinc',
          lemma: 'hinc',
          tipo: null,
          tabelle: null,
        },
        id: 'lessico_0_Hinc',
      }],
      lessicoRefs,
      hasLessico: true,
      toggleAperti,
    });

    expect(elementiGenerati).toEqual([expect.any(Object)]);

    // lessico 1
    expect(elementiGenerati[0].props).toEqual({
      children: 'Hi nc',
      onClick: expect.any(Function),
    });
    expect(elementiGenerati[0].key).toBe('lessico_block_lessico_0_Hinc');
    expect(elementiGenerati[0].type.displayName).toBe('Text__Lessico');
    expect(toggleAperti).not.toHaveBeenCalled();

    const LessicoStuff = () => elementiGenerati[0];
    expect(lessicoRefs.lessico_block_lessico_0_Hinc).toBeFalsy();
    const renderedComponent = mount(<LessicoStuff />);
    renderedComponent.simulate('click');
    expect(toggleAperti).toHaveBeenCalledWith({
      id: 'lessico_0_Hinc',
      inputLessicale: mockTesti[0].inputLessicale,
      parola: 'Hi_nc',
      type: 'lessico',
    });
    expect(lessicoRefs.lessico_block_lessico_0_Hinc).toBeTruthy();
  });
});


describe('<HtmlWrap />', () => {
  it('should render a Div', () => {
    const renderedComponent = shallow(<HtmlWrap />);
    expect(renderedComponent.find(Div).length).toEqual(1);
  });

  it('should render its css w/ props inline', () => {
    const renderedComponent = shallow(<HtmlWrap inline />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
