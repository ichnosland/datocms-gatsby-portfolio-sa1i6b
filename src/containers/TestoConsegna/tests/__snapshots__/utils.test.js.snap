// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlWrap /> should render its css w/ props inline 1`] = `
.c0 {
  width: 100%;
  display: inline;
}

.c0 .segnaposto-admin {
  display: none;
}

.c0 .consegna__titolo {
  display: block;
  padding-bottom: 20px;
  font-weight: 600;
}

.c0 .consegna__contenuto {
  color: #000;
}

.c0 .opzioni {
  margin-top: 10px;
  padding: 10px;
  background-color: rgba(96,141,212,0.1);
  border: 1px solid rgba(35,56,89,.1);
  border-radius: 2px;
}

.c0 .opzioni__titolo {
  display: block;
  padding-bottom: 10px;
}

.c0 .opzioni__contenuto {
  color: #000;
  display: inline-block;
  font-style: italic;
  padding-left: 6px;
}

.c0 .opzioni__contenuto .radiocheck-wrap {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0;
}

.c0 .opzioni .radio,
.c0 .opzioni .checkbox {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-right: 10px;
  padding-top: 3px;
}

.c0 .opzioni .radio:after,
.c0 .opzioni .checkbox:after {
  content: "";
  display: block;
  width: 14px;
  height: 14px;
  border: 1px solid;
  background: white;
}

.c0 .opzioni .checkbox:after {
  border-radius: 2px;
}

.c0 .opzioni .radio:after {
  border-radius: 100px;
}

.c0 .consegna__contenuto--esercizio {
  margin-top: 20px;
}

.c0 .completa {
  line-height: 2em;
}

.c0 .completa:after {
  content: "........................................";
  font-weight: 400;
}

.c0 .completa.full--width {
  display: block;
  height: 0;
  border-bottom: 1px dashed;
  margin: 2em 0;
}

.c0 .completa.full--width:after {
  content: "";
}

.c0 .completa.full--width:last-child {
  margin-bottom: 1em;
}

.c0 .mytestitaliano .contesto {
  border-bottom: 1px dotted;
}

.c0 .consegna__blocchi {
  margin-top: 10px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 4px;
  padding: 0 4px;
  border-radius: 2px;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco span {
  padding: 4px 0;
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza {
  border: 1px solid rgba(0,0,0,.4);
}

.c0 .consegna__blocchi .consegna__blocchi-blocco.analizza .segnaposto {
  border-bottom: 1px dashed rgba(0,0,0,.4);
}

.c0 p:first-child {
  margin-top: 0;
}

.c0 p:last-child {
  margin-bottom: 0;
}

.c0 img {
  max-width: 100%;
}

.c0 table {
  border-collapse: collapse;
  border-spacing: 0;
}

.c0 table th {
  background: rgb(231,236,236);
}

.c0 table td,
.c0 table th {
  padding: 0.62em;
  border: 1px solid rgb(231,236,236);
  vertical-align: middle;
}

.c0 .numero-esercizio {
  float: right;
  background-color: rgb(209,209,209);
  color: rgb(255,255,255);
  padding: 3px 6px;
  border-radius: 3px;
  text-align: right;
}

.c0 .editor {
  border: 1px dashed rgb(251,91,111);
}

.c0 .MathJax {
  font-size: 0.9em;
}

@media print {
  .c0 .opzioni {
    background: none;
  }
}

<Div
  className="c0"
  inline={true}
/>
`;
